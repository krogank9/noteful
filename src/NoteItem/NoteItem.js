import React from 'react';
import { Link } from 'react-router-dom';
import './NoteItem.css';
import NotefulContext from "../NotefulContext.js";

class App extends React.Component {
	static contextType = NotefulContext;
	
	static defaultProps = {
		note: {}
	};
	
	render() {
		return (
			<li className="noteItem">
				<h2>
					<Link to={"/note/"+this.props.note.id}>
						{this.props.note.name}
					</Link>
				</h2>
				<p>Date modified {this.props.note.modified}</p>
				<button
					onClick={() => this.context.requestDeleteNote(this.props.note.id)}
				>
					Delete note
				</button>
			</li>
		);
	}
}

export default App;
