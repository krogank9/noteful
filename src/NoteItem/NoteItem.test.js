import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteItem from './NoteItem'

describe(`NoteItem component`, () => {
	const props = {
		note: {
			id: 'a',
			name: 'test-class-name',
			modified: new Date(2018, 12, 15),
		}
	}

	it('renders a blank NoteItem by default', () => {
		const wrapper = shallow(<NoteItem />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders the NoteItem given props', () => {
		const wrapper = shallow(<NoteItem {...props} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
