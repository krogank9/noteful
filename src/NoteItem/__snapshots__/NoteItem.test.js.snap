// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem component renders a blank NoteItem by default 1`] = `
<li
  className="noteItem"
>
  <h2>
    <Link
      to="/note/undefined"
    />
  </h2>
  <p>
    Date modified 
  </p>
  <button>
    Delete note
  </button>
</li>
`;

exports[`NoteItem component renders the NoteItem given props 1`] = `
<li
  className="noteItem"
>
  <h2>
    <Link
      to="/note/a"
    >
      test-class-name
    </Link>
  </h2>
  <p>
    Date modified 
    <Component />
  </p>
  <button>
    Delete note
  </button>
</li>
`;
